### ISSUE TITLE
-cl-fp32-correctly-rounded-divide-sqrt  Algorithm XCASH ERROR

### ISSUE THREAD

**Issue by bycarloz** [fireice-uk/xmr-stak#2396]
![error1](https://user-images.githubusercontent.com/31080791/55589216-b2eb7200-56fd-11e9-9321-8f7d18d8b797.png)
I can not mine with the GPU ati radeon for that error. Algorithm XCASH ERROR

"currency" : "cryptonight_v8_double",

**Comment by bycarloz** [comment]
"currency" : "cryptonight_heavy"  The same happens with this same algorithm jumps this error.  -cl-fp32-correctly-rounded-divide-sqrt

**Comment by psychocrypt** [comment]
which gpu type do you use?

**Comment by bycarloz** [comment]
Ati 6870 1GB y ati 5970 2GB used the drivers 14.4, 15.7.1

**Comment by psychocrypt** [comment]
One if the gpus is to old and is not supporting a function needed by the
POW algorithm to create valid results.
We can not fix this issue (sry)

**Comment by bycarloz** [comment]
and can not delete -cl-fp32-right-rounded-divide-sqrt as in this case? Remove -cl-fp32-correctly-rounded-divide-sqrt from this line https://github.com/xmrig/xmrig-amd/blob/master/src/amd/OclCache.cpp#L72, it should fix build issue, but after this change algorithm cn/gpu will be broken. @psychocrypt

**Comment by psychocrypt** [comment]
yes correctly because cn_gpu requires to round floating point numbers
correctly.
I do not think that it is possible to fix it for your gpu.

**Comment by Spudz76** [comment]
That option turns on proper rounding as described by IEEE754

Older GPU didn't have good IEEE754 support.  Newer still sort of don't but at least support things like this rounding option to help keep things all calculating the same way.

If it rounds sqrt and div the "close enough" old way it will have too much error for this algo, but would be close enough literally for hand grenades (game math / physics)
